examples/views/dynamic_user_dropdown.py

```diff
@@ -0,0 +1,86 @@
+import discord
+
+bot = discord.Bot()
+
+# Defines a custom user specific dynamic dropdown
+# that the user can choose. The callback function
+# in the class is strictly to prevent the dreaded
+# interaction failed. The results are returned to
+# the calling function for further processing.
+
+
+def external_user_specific_list_function(discord_id):
+    # Replace this example with whatever your source is
+    # for the users dynamically generated items
+    users_menu_items = []
+    for i in range(5):
+        users_menu_items.append(
+            f"Item #{i + 1}) - Dynamically Generated for <@{discord_id}>"
+        )
+        i += 1
+    return users_menu_items
+
+
+class Dropdown(discord.ui.Select):
+    def __init__(self, interaction):
+        # In this example, we are passing in the interaction object from the calling interaction.
+        self.value = None
+        self.interaction = interaction
+        self.discord_user = interaction.user.id
+        self.users_menu_items = external_user_specific_list_function(self.discord_user)
+        options = [
+            discord.SelectOption(label=menu_item) for menu_item in self.users_menu_items
+        ]
+        # Label only in this example. But you could return a list of lists to populate other tags as well.
+        # i.e. for [discord.SelectOption(label=menu_item[0], description=menu_item[1]) for menu_item in self.users_menu_items]
+
+        # The placeholder is what will be shown when no option is selected.
+        # The min and max values indicate we can only pick one of the three options.
+        # The options parameter, contents shown above, define the dropdown options.
+        super().__init__(
+            placeholder="Choose your a menu item...",
+            min_values=1,
+            max_values=1,
+            options=options,
+        )
+
+    async def callback(self, interaction: discord.Interaction):
+        self.value = self.values[0]
+        await interaction.response.send_message(
+            content="Menu Item Selected...", ephemeral=True, delete_after=3
+        )  # could make 0 so it dissapears immediately
+        self.view.disable_all_items()
+        self.view.stop()
+
+
+# Defines a simple View that allows the user to use the Select menu.
+class DropdownView(discord.ui.View):
+    def __init__(self, interaction, *args, **kwargs) -> None:
+        super().__init__(Dropdown(interaction))
+
+        # Uses the one-liner from example in dropdown.py
+        # Passes the interaction object from calling interaction
+
+
+@bot.slash_command(name="dynamic_menu", description="Dynamic Menu Command")
+async def user_dynamic_menu(interaction: discord.Interaction):
```

Comment:
This is pycord, not discord.py
```suggestion
async def user_dynamic_menu(ctx: discord.ApplicationContext):
```

Reply:
I only use one library.  The need is to get the discord id into the menu factory.  If it can be done with the ApplicationContext as opposed to Interaction, then either works.